nxdrive/osi/linux/linux.py

```diff
@@ -100,3 +114,69 @@ def register_protocol_handlers(self) -> None:
             log.info(
                 f"Registered {original_executable!r} for URL scheme {NXDRIVE_SCHEME!r}"
             )
+
+    @if_frozen
+    def send_sync_status(self, state: DocPair, path: Path) -> None:
+        """
+        Send the sync status of a file to the FinderSync.
+
+        :param state: current local state of the file
+        :param path: full path of the file
+        """
+        try:
+            status = get_formatted_status(state, path)
+            if status:
+                log.debug(f"Sending status to FinderSync for {path!r}: {status}")
+                self._set_icon(status)
+        except Exception:
+            log.exception("Error while trying to send status to FinderSync")
+
+    def _set_icon(self, status: Dict[str, str]) -> None:
+        value = Status(int(status["value"]))
+        path = status["path"]
+        emblem = icon_status[value]
+        cmd = [
+            "gio",
+            "set",
+            "-t",
+            "stringv",
+            str(path),
+            "metadata::emblems",
+            icon_status[value],
+        ]
+        try:
+            subprocess.check_call(cmd)
+        except subprocess.CalledProcessError:
+            log.warning(f"Could not set the {emblem} emblem on {path!r}")
+        return
+
+    def _icons_to_emblems(self) -> None:
+        """
+            Copy nuxeo overlay icons to linux local icons folder.
+            Previous local icons will be replaced.
+        """
+        shared_icons = Path.home() / ".local/share/icons"
+        shared_icons.mkdir(parents=True, exist_ok=True)
+
+        status_list = ["synced", "syncing", "conflicted", "error", "locked", "unsynced"]
+        for status in status_list:
+
+            icon = find_icon("") / "overlay" / "linux" / ("badge_" + status + ".svg")
+            if not icon.exists():
+                continue
+            emblem = shared_icons / ("emblem-nuxeo_" + status + ".svg")
+
+            identical = False
+            try:
+                identical = filecmp.cmp(str(icon), str(emblem), shallow=False)
+            except Exception:
+                log.warning(f"Could not compare {icon!r} with {emblem!r}")
+            if emblem.is_file() and identical:
```

Comment:
This check should be moved before the identical one. Because if the emblem does not exist yet, the identical check will fail on a `FileNotFoundError`.